src/transform/import_resolution.rs

```diff
@@ -17,113 +16,133 @@ struct ImportsResolutionState<'a, R> {
+        match errors.first().cloned() {
+            Some(first) => Err(first),
+            None => Ok((term, resolved)),
+        }
     }
-}
 
-/// Resolve the import if the term is an unresolved import, or return the term unchanged. As
-/// [`super::share_normal_form::transform_one`], this function is not recursive.
-pub fn transform_one<R>(
-    rt: RichTerm,
-    resolver: &mut R,
-    parent: &Option<PathBuf>,
-) -> Result<RichTerm, ImportError>
-where
-    R: ImportResolver,
-{
-    let (rt, error) = transform_one_resilient(rt, resolver, parent);
-    match error {
-        Some(err) => Err(err),
-        None => Ok(rt),
+    /// Resolve the import if the term is an unresolved import, or return the term unchanged. As
+    /// [`super::share_normal_form::transform_one`], this function is not recursive.
+    pub fn transform_one<R>(
+        rt: RichTerm,
+        resolver: &mut R,
+        parent: &Option<PathBuf>,
+    ) -> Result<RichTerm, ImportError>
+    where
+        R: ImportResolver,
+    {
+        let (rt, error) = super::tolerant::transform_one(rt, resolver, parent);
+        match error {
+            Some(err) => Err(err),
+            None => Ok(rt),
+        }
     }
 }
 
-/// Performs imports resolution for each import term independent from others.
-/// It returns back a triple, containing:
-/// * A `RichTerm` which may still contain unresolved imports.
-/// * A `Vec<FileId>`, telling the imports which were resolved.
-/// * A `Vec<ImportError>`, telling the imports that couldn't be resolved.
-pub fn resolve_imports_resilient<R>(
-    rt: RichTerm,
-    resolver: &mut R,
-) -> (RichTerm, Vec<FileId>, Vec<ImportError>)
-where
-    R: ImportResolver,
-{
-    let mut stack = Vec::new();
-    let mut unresolved = Vec::new();
+/// Performs import resolution that cannot fail: it accumulates all errors and returns in
+/// along side with a (partially) resolved term.
+pub mod tolerant {
+    use super::ImportResolver;
+    use super::ImportsResolutionState;
+    use crate::error::ImportError;
+    use crate::term::{RichTerm, Term, Traverse, TraverseOrder};
+    use codespan::FileId;
+    use std::path::PathBuf;
 
-    let source_file: Option<PathBuf> = rt.pos.as_opt_ref().map(|x| {
-        let path = resolver.get_path(x.src_id);
-        PathBuf::from(path)
-    });
+    /// Performs imports resolution for each import term independent from others.
+    /// It returns back a triple, containing:
+    /// * A `RichTerm` which may still contain unresolved imports.
+    /// * A `Vec<FileId>`, telling the imports which were resolved.
+    /// * A `Vec<ImportError>`, telling the imports that couldn't be resolved.
+    pub fn resolve_imports<R>(
+        rt: RichTerm,
+        resolver: &mut R,
+    ) -> (RichTerm, Vec<FileId>, Vec<ImportError>)
+    where
+        R: ImportResolver,
+    {
+        let mut stack = Vec::new();
+        let mut unresolved = Vec::new();
```

Comment:
Small nitpick: `unresolved` makes it sound this is a vector of `FileId` for unresolved imports. I think it's rather a vector of `parse_errors`?